pkg/kapp/resources/resources.go

```diff
@@ -453,6 +453,14 @@ func (c *ResourcesImpl) isServerRescaleErr(err error) bool {
 	return false
 }
 
+// Retries retryable errors thrown by etcd server.
+// Comprehensive list of errors at : https://github.com/etcd-io/etcd/blob/main/server/etcdserver/errors.go
+// Addresses : https://github.com/vmware-tanzu/carvel-kapp/issues/106
+func (c *ResourcesImpl) isEtcdRetryableError(err error) bool {
+	return strings.Contains(err.Error(), "etcdserver:") &&
+		(strings.Contains(err.Error(), "timed out") || strings.Contains(err.Error(), "leader changed"))
```

Comment:
should we do this via a regexp?

Reply:
I think something as simple as `((etcdserver:).*(leader changed|timed out))` should work. Sounds like the better option. I will test it out and make the change